source/documentation/how-to/ha-vms.html.md

```diff
@@ -0,0 +1,84 @@
+
+Resume behavior can be selected on any VM, not just highly available VMs.  But there are special considerations regarding highly available VMs:
+
+- As long as a VM is paused, it may prevent Engine from starting it on another, working, host.
+- Or a paused VM may be started on another host and later resumed on the original host, resulting in split brain (the VM runs in multiple instances at the same time).
+
+There is also a special consideration if you use application level high availability in a non-highly available VM.  It may be a bad idea to use `AUTO_RESUME` resume behavior in such a case: If the VM gets paused and the application level high availability handles the situation by running the application elsewhere, confusion may arise if the paused VM starts again.  It may be better to use `LEAVE_PAUSED` or `KILL` resume behaviors in such setups.
+
+## VM leases
+
+VM lease is a feature that guarantees a VM can't be run in multiple instances at the same time.  That guarantee makes restarting highly available VMs safer and thus easier.  Split brain is not possible when a VM has been started with a lease and the lease is kept enabled for the VM.
+
+VM leases can be enabled only for highly available VMs, in High Availability section of the VM editing dialog again.  Just select *Target Storage Domain for VM Lease* in the dialog.
+
+Basically, a VM with a lease can start on a given host only after it acquires its lease there.  Once the VM stops, gets paused, is killed or its QEMU process crashes, it releases the lease, making it available for starting the VM on another host.  VM leases are thoroughly described in a separate [document about leases](/develop/release-management/features/storage/vm-leases/).
+
+To prevent split brain it's best to use VM leases on highly available VMs.  The same also helps restarting highly available VMs on other hosts in case of prolonged storage problems on some of the hosts.  Since paused VMs release their leases, the VMs become available for running on other hosts, but can cause trouble when attempting to resume them later on the original host.  For that reason `KILL` is the only possible resume behavior for VMs with leases, which helps destroy the paused VM instance when Engine moves the VM to Unknown status and attempts to start it elsewhere.  Of course, `KILL` behavior means killing the VM under the given conditions, so think about your setup before using it.
+
+## Additional notes
+
+VM resume behaviors are available since oVirt 4.2.  Older oVirt releases applied `AUTO_RESUME` behavior on all VMs.
+
+Up to oVirt 4.2.2 VMs with `KILL` behavior used to be killed only once the storage became available, i.e. Vdsm tried to resume the paused VMs.  Starting from oVirt 4.2.3 VMs are killed shortly after the timeout expires even when the storage problem persists.
```

Comment:
I think this should be optional setting, and should be a decision made by the engine - the host is not aware of the cluster's state that may allow us to do smarter (yes, and more complicated but not that much) things

Reply:
this article describes the current state. Such enhancements are to be documented/designed in RFEs

Reply:
I see, so 4.2.3 should be dropped then

Reply:
The change has already been merged to 4.2 so I updated the mentioned versions (and it indeed documents the current state).